app/javascript/pages/howToApply/how-to-apply.tsx

```diff
@@ -1,14 +1,28 @@
-import React, { ReactNode } from "react"
-
+import React, { ReactNode, useEffect, useState, useContext } from "react"
+import dayjs from "dayjs"
 import Layout from "../../layouts/Layout"
-import { Icon, t } from "@bloom-housing/ui-components"
+import type RailsSaleListing from "../../api/types/rails/listings/RailsSaleListing"
+import {
+  localizedFormat,
+  formatTimeOfDay,
+  getSfGovUrl,
+  renderInlineMarkup,
+  getPathWithoutLanguagePrefix,
+} from "../../util/languageUtil"
+import { Icon, t, NavigationContext, LoadingOverlay } from "@bloom-housing/ui-components"
+import { Message } from "@bloom-housing/ui-seeds"
 import withAppSetup from "../../layouts/withAppSetup"
-import { getSfGovUrl, renderInlineMarkup } from "../../util/languageUtil"
+import { getListing } from "../../api/listingApiService"
+
+import "./how-to-apply.scss"
 
 interface HowToApplyProps {
   assetPaths: unknown
 }
 
+const applicationsNotYetOpen = (listing: RailsSaleListing) =>
```

Comment:
There is a listing state function now: https://github.com/SFDigitalServices/sf-dahlia-web/blob/main/app/javascript/util/listingUtil.ts#L41